tools/shared.py

```diff
@@ -471,6 +471,10 @@ def find_temp_directory():
 def get_emscripten_version(path):
   return open(path).read().strip().replace('"', '')
 
+# Returns true if Emscripten is running in 'strict' mode, in which deprecated compiler features are not supported.
+def is_emscripten_strict():
+  return (os.environ.get('EMCC_STRICT') and int(os.environ.get('EMCC_STRICT')) != 0) or Settings.STRICT
```

Comment:
this can try to access `Settings.STRICT` before the `Settings` object exists, that worries me a lot.